cypress/e2e/product-reviews.cy.js

```diff
@@ -0,0 +1,34 @@
+// cypress/e2e/product-reviews.cy.js
+import ProductReviewsPage from '../support/page-objects/product-reviews-page'
+
+describe('Product Reviews', () => {
+    beforeEach(() => {
+        cy.visit('/search?q=shoes')
+    })
+
+    it('displays product cards with correct structure', () => {
+        ProductReviewsPage.verifyProductCard()
+    })
+
+    it('shows product details correctly', () => {
+        cy.get(ProductReviewsPage.selectors.productCard)
+            .first()
+            .within(() => {
+                // Verify product name
+                cy.get(ProductReviewsPage.selectors.productName)
+                    .should('be.visible')
+                    .and('not.be.empty')
+
+                // Verify shop name
+                cy.get(ProductReviewsPage.selectors.shopName)
+                    .should('be.visible')
+                    .and('not.be.empty')
+
+                // Verify product image
+                cy.get(ProductReviewsPage.selectors.productImage)
+                    .should('be.visible')
+                    .find('img')
+                    .should('have.attr', 'src')
```

Comment:
Any other options to verify the product image is there?

Reply:
Yes,  to ensure the image URL is valid and the image actually exists, we can use cy.request() to make an HTTP request to the image URL:

cy.get(ProductReviewPage.selectors.productImage)
  .should('be.visible')
  .invoke('attr', 'src')
  .then((src) => {
    cy.request(src).its('status').should('eq', 200);
  });


- can use the .invoke() command to check the src attribute directly.
cy.get(ProductReviewPage.selectors.productImage)
  .should('be.visible')
  .invoke('attr', 'src')
  .should('not.be.empty');